Mathlib/AlgebraicGeometry/EllipticCurve/Jacobian.lean

```diff
@@ -1068,6 +1100,27 @@ lemma nonsingular_neg {P : Fin 3 → F} (hP : W.Nonsingular P) : W.Nonsingular <
   · simp only [neg_of_Z_ne_zero hPz, nonsingular_smul _ <| Ne.isUnit hPz,
       nonsingular_neg_of_Z_ne_zero hP hPz]
 
+lemma addZ_neg {P : Fin 3 → R} : addZ P (W'.neg P) = 0 := by
+  simp_rw [addZ, neg, fin3_def_ext, sub_self]
+
+lemma addX_neg {P : Fin 3 → R} (hP : W'.Equation P) : W'.addX P (W'.neg P) = W'.dblZ P ^ 2 := by
+  simp only [addX, neg, dblZ, negY, fin3_def_ext]
+  linear_combination -2 * P z ^ 2 * (equation_iff _).mp hP
+
+lemma negAddY_neg {P : Fin 3 → R} (hP : W'.Equation P) :
+    W'.negAddY P (W'.neg P) = W'.dblZ P ^ 3 := by
+  simp only [negAddY, neg, dblZ, negY, fin3_def_ext]
+  linear_combination -2 * (2 * P y * P z ^ 3 + W'.a₁ * P x * P z ^ 4 + W'.a₃ * P z ^ 6)
+    * ((equation_iff _).mp hP)
+
+lemma addY_neg {P : Fin 3 → R} (hP : W'.Equation P) : W'.addY P (W'.neg P) = -W'.dblZ P ^ 3 := by
+  simp [addY, addX_neg hP, negAddY_neg hP, addZ_neg, negY, fin3_def_ext]
+
+lemma addXYZ_neg {P : Fin 3 → R} (hP : W'.Equation P) :
+    W'.addXYZ P (W'.neg P) = -W'.dblZ P • ![1, 1, 0] := by
+  simp_rw [addXYZ, addX_neg hP, addY_neg hP, addZ_neg, smul_fin3]
+  simp [(show Odd 3 by decide).neg_pow]
```

Comment:
No need for `simp`:
```suggestion
  erw [addXYZ, addX_neg hP, addY_neg hP, addZ_neg, smul_fin3, neg_sq, mul_one,
    Odd.neg_pow <| by decide, mul_one, mul_zero]
```
Now that we have this, can you also add the analogous statement `add_neg` after `add_of_X_ne`?